sklearn/model_selection/_validation.py

```diff
@@ -468,9 +472,17 @@ def _fit_and_score(estimator, X, y, scorer, train, test, verbose,
         # _score will return dict if is_multimetric is True
         test_scores = _score(estimator, X_test, y_test, scorer, is_multimetric)
         score_time = time.time() - start_time - fit_time
-        if return_train_score:
+        if return_train_score is True:
+            train_scores = _score(estimator, X_train, y_train, scorer,
+                                  is_multimetric)
+        if return_train_score is 'warn':
             train_scores = _score(estimator, X_train, y_train, scorer,
                                   is_multimetric)
+            score_train_time = time.time() - start_time - fit_time - score_time
+            if score_train_time >= 0.1*fit_time and time.time() - grid_search_start_time> 5:
```

Comment:
This still may mean that training score calculation time took only .5 of a second, which I don't think is quite enough. Maybe better would be to (naively) estimate overall training score time: `(time.time() - grid_search_start_time) * score_train_time / (fit_time + score_time + score_train_time) > 5`..?